docs/release_process.md

```diff
@@ -63,62 +63,20 @@ will be a new version of the Monitor web app on the [Production][prod]
 environment. To do this, we first release code to [Dev][dev] and
 [Stage][stage].
 
-## Release to Dev
+## Preview Deployment
```

Comment:
Awesome, thanks for updating that! The rest of this will need to be updated for CD but this is good for now